frontend/src/components/Meal/Day.js

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { DateTime } from "luxon";
+import Box from "@material-ui/core/Box";
+import Collapse from "@material-ui/core/Collapse";
+import IconButton from "@material-ui/core/IconButton";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from '@material-ui/core/Typography';
+import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
+
+import { APP_URL } from "../../config";
+import { getBackendAuthToken } from "../../session/localStorage";
+import { useDayStyles } from "../../styles/theme";
+
+const mealsTotalKcalSum = (meals) => {
+  return meals.reduce((prev, current) => prev + current.total_kcal, 0)
+}
+
+function getMealNameAsTime(mealDatetimeStr) {
+  var hour = Number(mealDatetimeStr.substr(11, 2));
+  var hourStr = mealDatetimeStr.substr(11, 5);
+  if (hour < 10) {
+    return "Breakfast at " + hourStr;
+  }
+  if (hour < 14) {
+    return "Lunch at " + hourStr;
+  }
+  if (hour < 17) {
+    return "Dinner at " + hourStr;
+  }
+  if (hour < 21) {
+    return "Supper at " + hourStr;
+  }
+  return "Night snack :( at " + hourStr;
+}
+
+const humanizeByDayOffset = (offset) =>
+  DateTime.local()
+    .minus({ days: offset })
+    .endOf("day")
+    .toLocaleString(DateTime.DATE_HUGE);
+
+const Day = ({ day }) => {
+  const [meals, setMeals] = useState([]);
+  const classes = useDayStyles();
+
+  useEffect(() => {
+    async function fetchMeals() {
+      const response = await axios.get(
+        APP_URL + "/api/meal/?dayOffset=" + day,
+        {
+          headers: {
+            Authorization: `Token ${getBackendAuthToken()}`,
+          },
+        }
+      );
+      setMeals(response.data);
+    }
+    fetchMeals();
+  }, []);
+
+  function IngredientDetails(item) {
+    const key = item.ingredient.id;
+    const name = item.ingredient.name;
+    const weight = item.weight;
+    const kcal_per_100g = item.ingredient.kcal_per_100g;
+    const kcal = item.kcal;
+    return (
+      <TableRow key={key}>
+        <TableCell>
+          {name}
+          <br></br>({kcal_per_100g} kcal)
+        </TableCell>
+        <TableCell align="right">{weight} g</TableCell>
+        <TableCell align="right">{kcal} kcal</TableCell>
+      </TableRow>
+    );
+  }
+
+  function Row({ name, total_kcal, ingredients }) {
+    const [open, setOpen] = React.useState(false);
+    const classes = useDayStyles();
```

Comment:
Should it be called `useMealRowStyles`? At least that what I thought was the pattern you were following?